src/som/interpreter/actors/EventualMessage.java

```diff
@@ -323,24 +331,40 @@ public static EventualMessage getCurrentExecutingMessage() {
 
   /**
    * Indicates that execution should stop and yield to the debugger,
-   * before the computed value is used to resolve the promise.
+   * before the message is processed.
+   */
+  public boolean haltOnReceive() {
+    return haltOnReceive;
```

Comment:
Is confusing because I have not seen it before. In Java is not common to use that naming you have used, that's why I sent you some links to JDK classes for you to see the pattern of `isPropertyName`. In your version, the only difference between variable and method name is the parenthesis.